app/lib/analyzer/backend.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math';
+
+import 'package:gcloud/db.dart';
+
+import '../shared/utils.dart';
+
+import 'models.dart';
+
+final Random _random = new Random.secure();
+
+class AnalysisBackend {
+  final DatastoreDB db;
+
+  AnalysisBackend(this.db);
+
+  Future<Analysis> getAnalysis(String package,
+      [String version, String analysis]) async {
+    final Key packageKey = db.emptyKey.append(PackageAnalysis, id: package);
+
+    Future<Analysis> loadRollup(AnalysisRollup rollup) async {
+      if (rollup == null) return null;
+      return (await db.lookup([rollup.analysisKey]))[0];
+    }
+
+    if (version == null) {
+      final List list = await db.lookup([packageKey]);
+      return loadRollup(list[0]);
+    }
+
+    // version was set
+    final Key versionKey =
+        packageKey.append(PackageVersionAnalysis, id: version);
+
+    if (analysis == null) {
+      final List list = await db.lookup([versionKey]);
```

Comment:
I will leave it this way, because:
- with `var`, the linter complains about being non-final
- without `List` the analyzer complains that there is a type missmatch